packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -69,9 +69,11 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         this.toDispose.push(Disposable.create(() => this.model = undefined));
 
         this.readPreferences();
-        this.toDispose.push(this.model.onDidChangeContent(() => this.readPreferences()));
-        this.toDispose.push(this.model.onDirtyChanged(() => this.readPreferences()));
-        this.toDispose.push(this.model.onDidChangeValid(() => this.readPreferences()));
+        if (this.model !== undefined) {
```

Comment:
How that can be undefined? it is assigned above.

Reply:
I had a few lint errors that failed the build stating that the model was possibly undefined, since it is defined as `MonacoEditorModel | undefined`. I agree that it is strange, since I did not get this lint error previously. Should I revert it even if it fails builds?

Reply:
yes, we need to fix linting instead

Reply:
Please revert monaco.d.ts file and this line. Here we make use of Monaco and need its typings in `@theia/preferences`. After that tsc error about `undefiend` will be gone as well.

Reply:
It was not addressed yet.